### DISCUSSION TITLE
Support for Vikrami calendar (Nepal)

### DISCUSSION THREAD

**Discussion by AayushKarki714** [gpbl/react-day-picker#2233] [Ideas]
Thank you for such awesome library. Is there any support for Nepali Calendar? It has exposed some formatters but It only works for formatting date picker. I want to be able to configure the internal logic of how a month is determined.

**Comment by gpbl** [comment]
@AbhiShake1 was looking for the same in https://github.com/gpbl/react-day-picker/issues/2212.

This is something we cannot add to DayPicker itself - it has to be supported by `date-fns`, or a library similar to `date-fns-jalali`.

See https://github.com/date-fns/date-fns/issues/1730, https://en.wikipedia.org/wiki/Vikram_Samvat

**Reply by AbhiShake1** [reply]
Thanks alot. I thought since the months/monthDays are different, adding ne locale to datefns wouldnt work. Seems like it does. Will find some free time and get on it

**Reply by gpbl** [reply]
@AbhiShake1 here is how we support the Jalali calendar:

https://github.com/gpbl/react-day-picker/blob/main/src/jalali.tsx

at first sight, creating a new npm package similar to [date-fns-jalal](https://github.com/date-fns-jalali/date-fns-jalali) could be the solution. Maybe its maintainer @smoosavi can help you understand the effort required?

**Reply by AbhiShake1** [reply]
> @AbhiShake1 here is how we support the Jalali calendar:
> 
> https://github.com/gpbl/react-day-picker/blob/main/src/jalali.tsx
> 
> at first sight, creating a new npm package similar to [date-fns-jalal](https://github.com/date-fns-jalali/date-fns-jalali) could be the solution. Maybe its maintainer @smoosavi can help you understand the effort required?

Thanks alot for the help. I want to try adding it to datefns first, then i may be able to understand why date-fns-jalali was created proceed onto a separate library. 
Thanks alot for helping out man, i will keep you posted
